src/server/frontend_wayland/xdg_shell_stable.cpp

```diff
@@ -573,22 +573,24 @@ void mf::XdgToplevelStable::send_toplevel_configure()
             *state = State::activated;
     }
 
-    switch (window_state())
+    if (auto const surface = scene_surface())
     {
-    case mir_window_state_maximized:
-    case mir_window_state_horizmaximized:
-    case mir_window_state_vertmaximized:
-        if (uint32_t *state = static_cast<decltype(state)>(wl_array_add(&states, sizeof *state)))
-            *state = State::maximized;
-        break;
+        auto const state = surface.value()->state_stack();
+        if (state.has(mir_window_state_maximized) ||
+            state.has(mir_window_state_horizmaximized) ||
+            state.has(mir_window_state_vertmaximized))
```

Comment:
Seeing this usage, maybe `has()` should take an `initializer_list<MirWindowState>`?

Reply:
Done